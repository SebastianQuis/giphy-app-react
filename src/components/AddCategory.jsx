import React, { useState } from 'react'

export const AddCategory = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState('');

    const onGiphyChanged = ( event ) => {
        // console.log( `ONCHANGED: ${event.target.value}` );
        setInputValue( event.target.value );
    }

    const onSubmitGphy = ( event ) => {
        event.preventDefault(); // FUNCIONAMIENTO BÁSICO
        if (inputValue.trim().length < 2) {
            alert('Tiene que ingresar más de 1 caracter');
            return;
        };
        onNewCategory( inputValue.trim() ) 
        setInputValue(''); // LIMPIANDO VALOR DE INPUT
    }

    return (
        <form onSubmit={ onSubmitGphy }>
            <input 
                type="text" 
                placeholder='Buscar gif'
                value={ inputValue } 
                onChange={ onGiphyChanged }
            />
        </form>
    )
}




