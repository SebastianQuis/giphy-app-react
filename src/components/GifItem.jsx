
import React from 'react'

import PropTypes from 'prop-types';



export const GifItem = ( { title, imgUrl } ) => {
    
    return (
        <div className="card" role='dialog'>
            <img src={ imgUrl } alt={ title } />
            <p>{ title }</p>
        </div>
    )

}


GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    imgUrl: PropTypes.string.isRequired
}

// GifItem.defaultProps = {
//     title: '',
//     imgUrl: '',
// }

