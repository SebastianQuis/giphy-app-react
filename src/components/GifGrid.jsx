
import { GifItem } from './GifItem';
import { UseFetchGifs } from '../hooks/UseFetchGifs';


export const GifGrid = ( {category} ) => {

  const { images, isLoading } = UseFetchGifs(category);

  // TODO - AGREGAR BOTON PARA LIMPIAR EL ARREGLO
  

  return (
    <>
      { isLoading && ( <h2>Cargando...</h2> ) }
      
      <div className="card-grid">
        { images.map( ( image ) => (
          <GifItem 
            key={ image.id } 
            { ...image }
          />
        ))}
      </div>
    </>
  )

}

