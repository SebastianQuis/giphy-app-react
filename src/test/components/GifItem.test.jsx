import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components";

describe('Pruebas en GifItem', () => { 

    const title = 'Saitama';
    const imgUrl = 'https://one-punch.com/saitame.jpg';

    test('funcion GifItem deberÃ­a arrojar: snapshot', () => { 
        const { container } = render(
            <GifItem imgUrl={imgUrl} title={title}/>
        );
        expect( container ).toMatchSnapshot();
    });



    // screen.debug(); // import react-testing-library

    test('should show the URL and ALT of the image', () => { 
        render( <GifItem imgUrl={imgUrl} title={title}/>);
        const { src, alt } = screen.getByRole('img');
        
        expect( src ).toBe( imgUrl );
        // expect( src ).toContain( "https://" );
        expect( alt ).toBe( title );
    });


    test('should show the title in the component <p> </p>', () => { 
        render( <GifItem imgUrl={imgUrl} title={title}/>);
        expect( screen.getByText(title) ).toBeTruthy();
        
        // TODO - segunda forma
        // const newTitle = screen.getByRole('paragraph').innerHTML;
        // expect( newTitle ).toBe( title );
    })
    
    test('should have the className CARD', () => { 
        render( <GifItem imgUrl={imgUrl} title={title}/>);
        const divElement = screen.getByRole('dialog');
        console.log(`123: ${divElement.className}`);
        
        // expect( divElement.className ).toBe('card');

    })


})