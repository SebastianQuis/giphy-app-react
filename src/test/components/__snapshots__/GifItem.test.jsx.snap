// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem funcion GifItem deber√≠a arrojar: snapshot 1`] = `
<div>
  <div
    class="card"
    role="dialog"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitame.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
